### DISCUSSION TITLE
Demo / solution  with any indicator working !  (my C# is rusty)

### DISCUSSION THREAD

**Discussion by jeonnagroup2** [DaveSkender/Stock.Indicators#578] [Q&A] [answered]
First this library look awesome. I'm a little rusty with my C# do you have a demo somewhere to look at it ! ?

i check the documentation but i don't remember exactly how IEnumerable work.

Ty for this hard work btw can't wait to understand the basic of it !!**

**Comment by DaveSkender** [comment, marked as answer]
First, check out the example in the guide:
https://dotnet.stockindicators.dev/guide/#example-usage

Then, for a basic Console App example:
https://dotnet.stockindicators.dev/examples/

For a more advanced Web API approach, take a look at the code for the demo site:
https://charts.stockindicators.dev

**Reply by DaveSkender** [reply]
The demo site is not the simplest form of a code example, so I’ll try to come up with a more basic working example, but will need some time to put it together.

In the meantime maybe someone else can provide some basic code examples or explanation.

**Reply by jeonnagroup2** [reply]
i understand that, maybe only a example of how "GetHistoryFromFeed" function could be implemented !!  I'm still reading on the subject i did my course in 2010 without working in the filed so my c# need some "refresh" hahah

**Reply by jeonnagroup2** [reply]
i made a few tests and i think i did it ! at least i get the historical data .  

![image](https://user-images.githubusercontent.com/91985268/136199842-8861657b-973f-491b-be84-29b7a88c71b5.png)

**Reply by DaveSkender** [reply]
Very good!  And yes, I'll also have a simple historical quote fetcher in my [future] example as I know many people coming here may want a full working example that's not as complicated as our demo site.

I've intentionally stayed out of quote fetching features with the library because I didn't want to have an opinionated or limited approach given how many data providers and use cases there are out there.

**Reply by jeonnagroup2** [reply]
we have to work !!  ty again for this awesome work !!

**Reply by DaveSkender** [reply]
You'll find that 100 bars may not be enough for some indicators, especially for those that have convergence warmup windows

**Reply by DaveSkender** [reply]
And thank you, for your [sponsorship](https://github.com/sponsors/DaveSkender), much appreciated! ❤️

**Reply by jeonnagroup2** [reply]
my pleasure Dave, i will upgrade it soon  !!  

btw i'm starting to understand the library and wow i really love it !!  Excellent work my friend !! 

![image](https://user-images.githubusercontent.com/91985268/136248670-dfd199f5-96a5-4fd2-aae9-a4a1df48b362.png)

**Reply by DaveSkender** [reply]
`Quote d = new Quote();` should probably be in your `for` loops, just a quick observation, I think.
Here's an approach I like:

```csharp

// transpose raw historical quotes
List<Quote> quotes = new();

for(...){

  Quote q = new()
      {
        Date = ...,
        Open = ...,
        High = ...,
        Low = ...,
        Close = ...,
        Volume = ...
      };

  quotes.Add(q);
}

// compute indicators
IEnumerable<SmaResult> smaHistory = quotes.GetSma(...);
```

**Reply by jeonnagroup2** [reply]
Done ty for the idea !!

**Reply by DaveSkender** [reply]
I've added a simple working console app example:
https://dotnet.stockindicators.dev/examples/
